include/mgmt/hawkbit/config_data.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2021 Advanced Climate Systems
+ *
+ * SPDX-License-Identiier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_MGMT_HAWKBIT_CONFIG_DATA_H_
+#define ZEPHYR_INCLUDE_MGMT_HAWKBIT_CONFIG_DATA_H_
+#include <autoconf.h>
+
+/**
+ * @brief Populate the config data struct.
+ *
+ * @param data pointer to the structure the device attributes should
+ * should be set on.
+ * @return 0 on succes, otherwise negative error code
+ */
+int hawkbit_get_config_data(struct hawkbit_cfg_data *data);
```

Comment:
this function should be in the existing `hawkbit.h` public header, or the public header should be moved into the 
new `hawkbit` folder

Reply:
as @ycsin said `hawkbit_get_config_data` function should be placed in the existing public header `hawkbit.h`

Reply:
My reasoning for putting it in a seperate header was this being a function that is internal to hawkBit. With my idea being that `hawkbit.h` would contain the public hawkbit API, and headers located in `subsys/mgmt/hawkbit/` would include optionally implemented functions, like in this case `hawkbit_get_config_data`, that are internally in hawkbit.

Apart from the above mentioned consideration there is also a technical one. `hawkbit_get_config_data` needs the definition of `struct hawkbit_config_data` in scope. This struct can defined in two ways depending on `CONFIG_HAWKBIT_CUSTOM_CONFIG_DATA`:

1. The default implementation is used. In this case (`CONFIG_HAWKBIT_CUSTOM_CONFIG_DATA=n`) `struct hawkbit_config_data` is defined in `zephyr/subsys/hawkbit/mgmt/hawkbit_config_data.h`. Only after this struct is defined, is the definition of `hawkbit_get_config_data` included because it depends on this struct having a definition, otherwise you get a compiler error.
2. The case where `CONFIG_HAWKBIT_CUSTOM_CONFIG_DATA=y`. Now the definition of `struct hawkbit_config_data` is in `include/mgmt/hawkbit/hawkbit_custom_data.h`, again this is now included `zephyr/subsys/hawkbit/mgmt/hawkbit_config_data.h`, before it includes the definition of `hawkbit_get_config_data`. Doing it this way prevents having to include `include/mgmt/hawkbit/hawkbit_custom_data.h` halfway through hawkbit.c.

This is the way I see it at least. Feel free to correct my way of thinking if I've got it wrong

Reply:
Sounds a bit complicated, my limited and simple understanding is that headers are considered public when they are in the `include` folder and are supposed to be included by user application to interact with the system applications. Real private headers should be within the same folder of `hawkbit.c` instead.

Reply:
would love to have your comments @gmarull